### PR TITLE
Update shape attribute references

### PR DESCRIPTION
**Pull request by nickmcintyre**
Preparing references in `src/core/shape/attributes.js` for the new website. I also took a crack at addressing #5472.

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated

@Qianqianye @davepagurek @limzykenneth

### CODE REVIEW

File: src/core/shape/attributes.js [hunk 1]
```diff
@@ -10,50 +10,77 @@ import p5 from '../main';
 import * as constants from '../constants';
 
 /**
- * Modifies the location from which ellipses, circles, and arcs are drawn. By default, the
- * first two parameters are the x- and y-coordinates of the shape's center. The next
- * parameters are its width and height. This is equivalent to calling `ellipseMode(CENTER)`.
+ * Changes the way ellipses, circles, and arcs are drawn.
```
Comment (Qianqianye) [inline src/core/shape/attributes.js:13]: How about 'Change where ellipses, circles, and arcs are drawn'? @nickmcintyre

File: src/core/shape/attributes.js [hunk 2]
```diff
@@ -71,26 +98,75 @@ p5.prototype.ellipseMode = function(m) {
 };
 
 /**
- * Draws all geometry with jagged (aliased) edges.
+ * Draws certain features with jagged (aliased) edges.
```
Comment (Qianqianye) [inline src/core/shape/attributes.js:101]: Could you share the reason to make the change from 'Draws all geometry' to 'Draws certain features'? It seems make more sense to 'draw geometry' than 'draw features'. If it's important to keep 'certain', how about 'Draws certain geometry'? @nickmcintyre

File: src/core/shape/attributes.js [hunk 3]
```diff
@@ -106,55 +182,108 @@ p5.prototype.noSmooth = function() {
 };
 
 /**
- * Modifies the location from which rectangles and squares are drawn. By default,
- * the first two parameters are the x- and y-coordinates of the shape's upper-left
- * corner. The next parameters are its width and height. This is equivalent to
- * calling `rectMode(CORNER)`.
+ * Changes the way rectangles and squares are drawn.
```
Comment (Qianqianye) [inline src/core/shape/attributes.js:185]: Similar comment here, How about 'Changes where rectangles and squares are drawn'? @nickmcintyre

File: src/core/shape/attributes.js [hunk 4]
```diff
@@ -172,27 +301,76 @@ p5.prototype.rectMode = function(m) {
 };
 
 /**
- * Draws all geometry with smooth (anti-aliased) edges. <a href="#/p5/smooth">smooth()</a> will also
- * improve image quality of resized images.
+ * Draws certain features with smooth (antialiased) edges.
```
Comment (Qianqianye) [inline src/core/shape/attributes.js:304]: similar comment as 'noSmooth'
Could you share the reason to make the change from 'Draws all geometry' to 'Draws certain features'? It seems make more sense to 'draw geometry' than 'draw features'. If it's important to keep 'certain', how about 'Draws certain geometry'? @nickmcintyre

File: src/core/shape/attributes.js [hunk 5]
```diff
@@ -312,39 +538,57 @@ p5.prototype.strokeJoin = function(join) {
 };
 
 /**
- * Sets the width of the stroke used for lines, points, and the border around
- * shapes. All widths are set in units of pixels.
+ * Sets the width of the stroke used for points, lines, and the border around
```
Comment (Qianqianye) [inline src/core/shape/attributes.js:541]: This is related to the 'stroke()' reference and the [comment](https://github.com/processing/p5.js/pull/6839#discussion_r1518993692) in #6839 

Is it important to keep both 'lines' and 'border'? If so, we should keep the consistency to referring border, which is currently called 'outline' in stroke() reference. 
@limzykenneth @davepagurek

File: src/core/shape/attributes.js (hunk 2 above)
Comment (davepagurek) [reply src/core/shape/attributes.js:101]: I think the reason for this is that there's no way to turn off antialiasing on lines and shapes in 2D mode, and I think this confuses a lot of users (it's come up in the discord a few times.) In 2D mode, I believe this just affects image scaling in 2D mode, so it actually doesn't affect geometry at all, only images. Somewhat confusingly, this *does not* affect image scaling in WebGL mode (you have to use an [undocumented method on p5.Texture](https://github.com/processing/p5.js/blob/7a3d73bba4214dab2319e18c8efa4aaa5db3d010/src/webgl/p5.Texture.js#L337)) but *does* actually affect antialiasing. The situation is unfortunately a bit complicated right now :')

I think the description in this PR does a pretty good job of mentioning the key details quickly.

File: src/core/shape/attributes.js (hunk 5 above)
Comment (davepagurek) [reply src/core/shape/attributes.js:541]: I think changing `border` to `outline` here makes sense! I'm also in favour of keeping points and lines to make clear what this gets used for.

File: src/core/shape/attributes.js (hunk 1 above)
Comment (nickmcintyre) [reply src/core/shape/attributes.js]: Updated!

File: src/core/shape/attributes.js (hunk 3 above)
Comment (nickmcintyre) [reply src/core/shape/attributes.js]: Updated!

File: src/core/shape/attributes.js (hunk 2 above)
Comment (nickmcintyre) [reply src/core/shape/attributes.js:101]: @Qianqianye basically what @davepagurek said. I sketched a few test cases based on #5472 and tried to describe what I saw.

File: src/core/shape/attributes.js (hunk 5 above)
Comment (nickmcintyre) [reply src/core/shape/attributes.js]: Updated!

**Pull request merged**